app/models/rubric_criterion.rb

```diff
@@ -136,8 +144,10 @@ def self.load_from_yml(criterion_yml)
     criterion.name = name
     criterion.max_mark = criterion_yml[1]['max_mark']
 
+    # get number of required fields
+    required_fields = Level.validators.grep(ActiveRecord::Validations::PresenceValidator).length
     # Next comes the level names.
-    (0..RUBRIC_LEVELS - 1).each do |i|
+    (0..required_fields - 1).each do |i|
```

Comment:
This change is incorrect (you're mixing up the number of levels a criterion has by default with the number of required fields for a single level). Please revert this change and instead as @YolandaZIK to handle this, as she is working on this method.